src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -457,6 +457,44 @@ bool DxFontRenderData::DidUserSetFeatures() const noexcept
     return _didUserSetFeatures;
 }
 
+// Routine Description:
+// - Returns whether the user set or updated any of the font axes to be applied
+bool DxFontRenderData::DidUserSetAxes() const noexcept
+{
+    return _didUserSetAxes;
+}
+
+// Routine Description:
+// - Function called to inform us whether to use the user set weight
+//   in the font axes
+// - Called by CustomTextLayout, when the text attribute is bold we should
+//   ignore the user set weight, otherwise setting the bold font axis
+//   breaks the bold font attribute
+// Arguments:
+// - useUserWeight: boolean that tells us if we should use the user set weight
+//   in the font axes
+void DxFontRenderData::UseUserWeight(bool useUserWeight) noexcept
+{
+    _useUserWeight = useUserWeight;
+}
+
+// Routine Description:
+// - Returns whether the set italic in the font axes
+// Return Value:
+// - True if the user set the italic axis to 1,
+//   false if the italic axis is not present or the italic axis is set to 0
+bool DxFontRenderData::DidUserSetItalic() const noexcept
+{
+    for (const auto& axisVal : _axesVector)
+    {
+        if (axisVal.axisTag == DWRITE_FONT_AXIS_TAG_ITALIC && axisVal.value == 1)
+        {
+            return true;
+        }
+    }
+    return false;
+}

```

Comment:
Consider eagerly computing and then caching this when the `_axesVector` is changed.

Reply:
That's probably better yeah, thanks!